Mongo integration

limonana opened the pull request:
create InitDB application to fill mongo with initial data
implement  getMovieById to query the db

limonana committed:
add endpoint InitDB and chnage GetMovieById to use mongo

limonana committed:
order Init code

limonana committed:
work with default db

limonana committed:
empty movies collection

limonana committed:
fix getMovieById to use string

limonana committed:
extract connect to DB to different module

limonana committed:
refactor initDB to be written with async await

limonana committed:
refactor getMovieById to use async await

limonana committed:
getMovieById use _id instead of id

limonana committed:
initDB use mongo _id

limonana committed:
create initDB app

yarindeoh commented on .vscode/launch.json:
```diff
@@ -0,0 +1,14 @@
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
```
is this part of the IDE conf ? 
if so, we can insert it to gitignore

limonana replied:
yes. I forgot about this option

limonana committed:
ignore launch.json

limonana committed:
delete launch.json

limonana committed:
merge from master

limonana committed:
replace movieDB with movieRepo
comment out all api calls beside /:movieId in order not implment unneeded functionality in MoviesRepo

limonana committed:
order require

limonana committed:
create db named rate-movies

limonana committed:
add documantion about using mongo

limonana commented:
@yarindeoh sorry but I commented out unneeded CRUD operation as we agree they are not needed right now and I didn't want to  implement them if not needed.

limonana commented:
something I didn't succeed figure out is what are the best practices connecting to mongo. 
Should I save the Db object returned from dbServer.db('rate-movies') and use it always or is it OK the way it is done now. because I didn't succeed in figuring this out, I left the code as it because it is simpler.

limonana committed:
share same DB connection across server

limonana committed:
exit server when can't connect to DB

limonana committed:
close the connection when application exit

limonana committed:
use useNewUrlParser to avoid warning

limonana committed:
change var to let
